File: ECommerce.Domain.Model/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ECommerce.Domain.Model.Models
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Slug { get; set; }

        public string Thumbnail { get; set; }

        public string ShortDescription { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }
    }
}

File: ECommerce.WebApp/ClientApp/models/products/product.ts

export interface IProductModel {
    Id: number;
    Name: string;
    Slug: string;
    Thumbnail: string;
    ShortDescription: string;
    Description: string;
    Price: number;
}

export class ProductModel implements IProductModel {
    public Id: number = 0;
    public Name: string = null as any;
    public Slug: string = null as any;
    public Thumbnail: string = null as any;
    public ShortDescription: string = null as any;
    public Description: string = null as any;
    public Price: number = 0;
}